primitives/src/lib.rs

```diff
@@ -102,3 +103,16 @@ impl TryFrom<Vec<u8>> for CurrencyId {
 		}
 	}
 }
+
+/// A trait that allows version migrators to access the underlying pallet's
+/// context, e.g., its Config trait.
+///
+/// In this way, the migrator can access the pallet's storage and the pallet's
+/// types directly.
+pub trait VersionMigratorTrait<T> {
```

Comment:
Not sure whether we want to introduce a new `Trait` pallet for this instead.

Reply:
Perhaps it would make sense. But at the moment `kilt_primitives` was the only place I could put it in without creating any dependencies between KILT pallets. I guess also the storage migration stuff deserves to be put in a different crate, perhaps the same as the one for crates.

Reply:
Yes. I am fine leaving it be for this PR and then we handle it correctly in another PR.